save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/projects/ProjectSettingsMenu.kt

```diff
@@ -82,18 +84,56 @@ private fun projectSettingsMenu() = FC<ProjectSettingsMenuProps> { props ->
     @Suppress("LOCAL_VARIABLE_EARLY_DECLARATION")
     val projectRef = useRef(props.project)
     val (draftProject, setDraftProject) = useState(props.project)
-
     useEffect(props.project) {
         if (projectRef.current !== props.project) {
             setDraftProject(props.project)
             projectRef.current = props.project
         }
     }
+    val navigate = useNavigate()
 
     val projectPath = props.project.let { "${it.organization.name}/${it.name}" }
+    val deleteProject = useDeferredRequest {
+        val responseFromDeleteProject = delete(
+            "$apiUrl/projects/$projectPath/delete",
+            jsonHeaders,
+            body = undefined,
+            loadingHandler = ::noopLoadingHandler,
+        )
+        if (responseFromDeleteProject.ok) {
+            navigate("/${FrontendRoutes.PROJECTS}")
+        }
+    }
 
-    val (wasConfirmationModalShown, showGlobalRoleWarning) = useGlobalRoleWarningCallback(props.updateNotificationMessage)
+    val updateProject = useDeferredRequest {
+        post(
+            "$apiUrl/projects/update",
+            jsonHeaders,
+            Json.encodeToString(draftProject.toDto()),
+            loadingHandler = ::loadingHandler,
+        ).let {
+            if (it.ok) {
+                props.onProjectUpdate(draftProject)
+            }
+        }
+    }
+
+    val (isModalOpen, setIsModalOpen) = useState(false)
```

Comment:
suggestion to use `useWindowOpenness`

Reply:
oh I do not like that hook :(

Reply:
instead of general state with boolean value, it provides an object with methods with meaningful names like `isOpen`, `openWindow`/`closeWindow` and `openWindowAction`\`closeWindowAction`

Reply:
It's more about ***over***loading imo🤓
But if you insist, I will do it

Reply:
Done